src/__tests__/pet.test.js

```diff
@@ -0,0 +1,168 @@
+    expect (fido.hunger).toEqual (5);
+  });
+
+  test ('has an initial fitness of 10', () => {
+    expect (fido.fitness).toEqual (10);
+  });
+
+  test ('decrements the fitness by 3', () => {
+    fido.growUp();
+    expect (fido.fitness).toEqual (7);
+  });
+
+  test(`growUp should throw error: "${ERROR_MESSAGE}"`, () => {
+    fido.age = 30;
+    expect(fido.isAlive).toBe(false);
+    expect(() => {
+        fido.growUp();
+    }).toThrowError(new Error(ERROR_MESSAGE));
+  });
+});
+  
+describe ('walk', () => {
+  test('fitness cannot go above 10', () => {
+    fido.fitness = 10;
+    fido.walk();
+    expect(fido.fitness).toBe(10);
+  });
+
+  test('increments the fitness property by 4', () => {
+    fido.fitness = 5;
+    fido.walk();
+    expect(fido.fitness).toBe(9);
+  });
+
+  test(`walk should throw error: "${ERROR_MESSAGE}"`, () => {
+    fido.fitness = 0;
+    expect(fido.isAlive).toBe(false);
+    expect(() => {
+        fido.walk();
+    }).toThrowError(new Error(ERROR_MESSAGE));
+  });
+});
+   
+describe ('feed', () => {
+
+  test ('hunger cannot go under 0', () => {
+    fido.hunger = 2;
+    fido.feed();
+    expect (fido.hunger).toEqual(0);
+  });
+
+  test ('decrements the hunger by 3', () => {
+    fido.hunger = 4;
+    fido.feed();
+    expect (fido.hunger).toEqual (1);
+  });
+
+  test(`feed should throw error: "${ERROR_MESSAGE}"`, () => {
+    fido.fitness = 0;
+    expect(fido.isAlive).toBe(false);
+    expect(() => {
+        fido.feed();
+    }).toThrowError(new Error(ERROR_MESSAGE));
+  });
+});
+
+describe ('checkUp', () => {
+  test('return "I need a walk" if fitness is 3 or less', () => {
+    fido.fitness = 3;
+    expect(fido.checkUp()).toEqual('I need a walk');
+  });
+
+  test('return "I am hungry" if hunger is 5 or more', () => {
+    fido.hunger = 6;
+    expect(fido.checkUp()).toEqual('I am hungry');
+  });
+
+  test('return "I am hungry AND I need a walk" if hunger is 5 or more AND fitness is 3 or less', () => {
+    fido.fitness = 2;
+    fido.hunger = 6;
+    expect(fido.checkUp()).toEqual('I am hungry AND I need a walk');
+  });
+  test('return "I feel great!" if none of the above are true', () => {
+    expect(fido.checkUp()).toEqual('I feel great!');
+   });
+});
+
+describe ('isAlive', () => {
+  test('return false if fitness is 0 or less', () => {
+    fido.fitness = 0;
+    expect(fido.isAlive).toBe(false);
+  });
+
+  test('return false if hunger is 10 or more', () => {
+    fido.hunger = 10;
+    expect(fido.isAlive).toBe(false);
+  });
+
+  test('return false if age is 30 or more', () => {
+    fido.age = 30;
+    expect(fido.isAlive).toBe(false);
+  });
+  test('return true if none of the above are true', () => {
+    expect(fido.isAlive).toBe(true);
+  });
+});
+
+describe ('haveBaby', () => {
+
+  test('creates a child pet from inside its parent pet', () => {
+    const parent = new Pet ('Tee');
+    parent.haveBaby('Mo');
+    expect(parent.children).toBeInstanceOf(Array);
```

Comment:
I would also check that the name of the baby in the array is correct as part of this test as even if the array is empty, it will still be an instance of an array.